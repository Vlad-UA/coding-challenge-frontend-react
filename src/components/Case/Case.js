// Core
import React, { PureComponent } from 'react';
import moment from 'moment';
import callApi from '../../utils/call-api';

// Instruments
import styles from './Case.module.scss';

// Components
import Header from '../Header/Header';

export default class Case extends PureComponent {
  state = {
    incident: {},
  };

  async componentDidMount() {
    const { match: { params: { caseId } }, location: { state: { incident } = {} } } = this.props;

    if (incident) {
      this.setState({ incident });
    } else {
      try {
        const { incident } = await callApi({ id: caseId });
        console.log('NEW incident', incident);
        this.setState({ incident });
      } catch (error) {
        throw new Error(`ERROR call api with caseId : ${caseId}`);
      }
    }
  }

  render() {
    const { incident: { title, updated_at, description, address } } = this.state;

    console.log('this.props', this.props);

    return (
      <div className={styles.Case}>
        <Header />
        <div>{title}</div>
        <div>{`Stolen ${moment.unix(updated_at).format('llll')} at ${address}`}</div>
        <h2>DESCRIPTION OF INCIDENT</h2>
        <div>{description}</div>
      </div>
    );
  }
}
