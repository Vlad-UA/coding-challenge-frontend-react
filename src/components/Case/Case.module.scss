.Case {
  margin: 50px;
}
