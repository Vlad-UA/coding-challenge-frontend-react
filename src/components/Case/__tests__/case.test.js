// Core
import React from 'react';

// Test utilities
import renderer from 'react-test-renderer';

// Component
import Case from '../Case';

describe('Component Case:', () => {
  test('Must match snapshot:', () => {
    const caseComponent = renderer
      .create(<Case />)
      .toJSON();

    expect(caseComponent).toMatchSnapshot();
  });
});
