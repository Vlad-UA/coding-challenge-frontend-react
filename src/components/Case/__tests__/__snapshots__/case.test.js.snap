// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Case: Must match snapshot: 1`] = `
<div
  className="Case"
>
  <div
    className="Header"
  >
    <img
      alt="logo"
      src="logo.png"
    />
    <div>
      <h1>
        Police Department of Berlin
      </h1>
      <h2>
        Stolen bikes
      </h2>
    </div>
  </div>
  <div />
  <div>
    Stolen Invalid date at undefined
  </div>
  <h2>
    DESCRIPTION OF INCIDENT
  </h2>
  <div />
</div>
`;
